src/repositories/commits/commit.service.ts

```diff
@@ -25,12 +28,23 @@ export class CommitService {
     pageNumber = 1,
   ) {
     const commits: Commit[] = await queryCommits(repoIdent, pageNumber);
+    const commitDocuments: CommitDocument[] = [];
+    for (const commit of commits) {
+      let author: UserDocument;
+      if (commit.author) {
+        author = await retrieveDocument(this.userModel, {
+          id: commit.author.id,
+        });
+        if (!author) {
+          author = await createDocument(this.userModel, commit.author);
+        }
```

Comment:
I really really dislike this pattern.
I get what you are doing here, but I would wish it to be a bit more explicit.
if(author.exists) {commit.author = existingAuthor.id} else {createAuthor}
assigning twice to the author variable and using its value as a check for a conditional new initialization reads a bit odd in my opinion

Reply:
I see your point. But I don't think there is a way to avoid the conditional assignment to author. 
```ts
      if (commit.author) {
        if (await documentExists(this.userModel, { id: commit.author.id })) {
          author = await retrieveDocument(this.userModel, { id: commit.author.id })
        } else {
          author = await this.userModel.create(commit.author);
        }
        commit.author = author;
     }
```
Having to pass the same filter twice is a bit cumbersome, though. A more elegant solution would be using try/catch in `retrieveDocument`:
```ts
export async function retrieveDocument<T extends Document>(
  model: Model<T>,
  filter: FilterQuery<T>,
): Promise<T> {
  if (!documentExists(model, filter)) {
    throw new Error(`No such document: ${filter}`);
  }
  return model.findOne(filter).exec();
}
```
And in `storeCommits`:
```ts
      if (commit.author) {
        try {
          author = await retrieveDocument(this.userModel, { id: commit.author.id });
        } catch (e) {
          author = await this.userModel.create(commit.author)
        }
        commit.author = author
      }
```

Reply:
yeah the duplicate assignment would not be resolved but it would be more explicit what is going on there.
in the previous version `if (author) {...} ` the type of author is not so explicit. because if (author) looks like a bool. I know it is a check for undefined, but I find it a bit confusing.